honeybee_3dm/face.py

```diff
@@ -0,0 +1,145 @@
+import rhino3dm
+import uuid
+from togeometry import to_face3d
+from tojson import publish_json
+
+try:  # import the core honeybee dependencies
+    from honeybee.face import Face
+    from honeybee.shade import Shade
+    from honeybee.aperture import Aperture
+    from honeybee.door import Door
+    from honeybee.facetype import face_types
+    from honeybee.boundarycondition import boundary_conditions
+    from honeybee.typing import clean_and_id_string
+    from honeybee.facetype import face_types
+except ImportError as e:
+    raise ImportError('\nFailed to import honeybee:\n\t{}'.format(e))
+
+try:
+    from ladybug_geometry.geometry3d.face import Face3D
+
+except ImportError as e:
+    raise ImportError('\nFailed to import ladybug_geometry:\n\t{}'.format(e))
+
+
+try:  # import the honeybee-energy extension
+    from honeybee_energy.lib.constructions import opaque_construction_by_identifier
+except ImportError as e:
+    if len(ep_constr_) != 0:
+        raise ValueError('ep_constr_ has been specified but honeybee-energy '
+                         'has failed to import.\n{}'.format(e))
+
+try:  # import the honeybee-radiance extension
+    from honeybee_radiance.lib.modifiers import modifier_by_identifier
+except ImportError as e:
+    if len(rad_mod_) != 0:
+        raise ValueError('rad_mod_ has been specified but honeybee-radiance '
+                         'has failed to import.\n{}'.format(e))
+
+
+# The path here is the path to the rhinofile in .3dm format
+path = "D:\Github\Work-in-Progress\RhinoToHBJSON\sample.3dm"
+file = rhino3dm.File3dm.Read(path)
+
+
+# Honeybee faces will be collected here
+hb_faces = []
+
+# Gathering layer information from the rhino file
+layers = file.Layers
+# Layer names in the Rhino file
+layer_names = [layer_name.Name for layer_name in layers]
+# Layer Indexes for the layers in the Rhino file
+layer_indexes = [layer_name.Index for layer_name in layers]
+# A dictionary with the layer names as keys and layer indexes as values
+layer_dict = dict(zip(layer_names, layer_indexes))
+
+
+def layer_name_to_hb_objects(layer):
+    """This function takes a Rhino layer and spits out a Honeybee face type
+    and Honeybee Class to be used for Rhino to Honeybee translation.
+
+    Args:
+        layer (A string): Name of the Rhino layer
+
+    Returns:
+        hb_type: A Honeybee face type object
+        hb_face_module: A Honeybee class. It is either of Face, Shade, Aperture, or Door
+    """
+    hb_type = " "  # Variable holding a Honeybee face type
+    hb_face_module = " "  # Variable holding a Honeybee module name
```

Comment:
Why do you need this? It will only happen in the final `else` check. I would just use it there.

Reply:
These are variables created within the scope of the function. If I don't declare them, it doesn't work. The other way is to declare them as `global` variables, which I don't think we should do.